streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -172,16 +172,12 @@ void registerStateStores() {
         log.trace("Initializing state stores");
 
         for (final StateStore store : topology.stateStores()) {
-            log.debug("Initializing store {}", store.name());
             processorContext.uninitialize();
             store.init(processorContext, store);
+            log.debug("Initialized state store {}", store.name());
         }
     }
 
-    void reinitializeStateStoresForPartitions(final Collection<TopicPartition> partitions) {
```

Comment:
The re-init logic inside task / state manger is when a restoring process hit an OutOfRangeException from consumer, this happens when e.g. the changelog was truncated at the time, and we just want to re-init the restoration inside the task at the lower level.

In the past we have seen quite lot of messiness and bugs because of this logic, and I've decided to remove it and instead push it to higher-level thread to decide what to do: e.g. it can close it as unclean and re-create it again, in which case the initialization / restoration would be executed. I think it is much cleaner in this way.

The thread-level change would not be in this PR so we just remove it here.